src/main/java/io/axoniq/demo/university/faculty/automation/allcoursesfullybookednotifier/WhenAllCoursesFullyBookedThenSendNotification.java

```diff
@@ -0,0 +1,148 @@
+ */
+public class WhenAllCoursesFullyBookedThenSendNotification {
+
+    private static final String FACULTY_ID = "ONLY_FACULTY_ID";
+
+    @EventSourcedEntity
+    record State(Map<CourseId, Course> courses, boolean notified) {
+
+        record Course(int capacity, int students) {
+
+            Course capacity(int newCapacity) {
+                return new Course(newCapacity, this.students);
+            }
+
+            Course studentSubscribed() {
+                return new Course(this.capacity, this.students + 1);
+            }
+
+            Course studentUnsubscribed() {
+                return new Course(this.capacity, this.students - 1);
+            }
+
+            boolean isFullyBooked() {
+                return students >= capacity;
+            }
+
+        }
+
+        @EntityCreator
+        State() {
+            this(new HashMap<>(), false);
+        }
+
+        @EventSourcingHandler
+        State evolve(CourseCreated event) {
+            courses.put(event.courseId(), new Course(event.capacity(), 0));
+            return new State(courses, notified);
+        }
+
+        @EventSourcingHandler
+        State evolve(CourseCapacityChanged event) {
+            courses.computeIfPresent(event.courseId(), (id, course) -> course.capacity(event.capacity()));
+            return new State(courses, notified);
+        }
+
+        @EventSourcingHandler
+        State evolve(StudentSubscribedToCourse event) {
+            courses.computeIfPresent(event.courseId(), (id, course) -> course.studentSubscribed());
+            return new State(courses, notified);
+        }
+
+        @EventSourcingHandler
+        State evolve(StudentUnsubscribedFromCourse event) {
+            courses.computeIfPresent(event.courseId(), (id, course) -> course.studentUnsubscribed());
+            return new State(courses, notified);
+        }
+
+        @EventSourcingHandler
+        State evolve(AllCoursesFullyBookedNotificationSent event) {
+            return new State(courses, true);
+        }
+    }
+
+    static class AutomationCommandHandler {
+
+        @CommandHandler
+        public void decide(
+                SendAllCoursesFullyBookedNotification command,
+                @InjectEntity(idProperty = FacultyTags.FACULTY_ID) State state,
+                ProcessingContext context
+        ) {
+            var canNotify = state != null && !state.notified();
+            if (canNotify) {
+                var notification = new NotificationService.Notification("admin", "All courses are fully booked now.");
+                context.component(NotificationService.class).sendNotification(notification);
+                var eventAppender = EventAppender.forContext(context);
+                eventAppender.append(new AllCoursesFullyBookedNotificationSent(command.facultyId()));
+            }
+        }
+    }
+
+    static class AutomationEventHandler {
+
+        @EventHandler
+        public MessageStream.Empty<?> react(
```

Comment:
As pointed out on the documentation PR, I don't fully comprehend why you went for the `MessageStream.Empty<?>` if you can just do `void`.

Or, perhaps better, return the `CompletableFuture` from using the `CommandGateway`!